cmd/buildah/manifest.go

```diff
@@ -555,31 +556,52 @@ func manifestInspectCmd(c *cobra.Command, args []string, opts manifestInspectOpt
 		return errors.Wrapf(err, "error building system context")
 	}
 
-	ref, _, err := util.FindImage(store, "", systemContext, imageSpec)
+	refs, err := util.ResolveNameToReferences(store, systemContext, imageSpec)
 	if err != nil {
-		if ref, err = alltransports.ParseImageName(imageSpec); err != nil {
-			return err
-		}
+		return errors.Wrapf(err, "error resolving image names")
+	}
+
+	if ref, _, err := util.FindImage(store, "", systemContext, imageSpec); err == nil {
+		refs = append(refs, ref)
+	} else if ref, err := alltransports.ParseImageName(imageSpec); err == nil {
+		refs = append(refs, ref)
 	}
 
 	ctx := getContext()
+	var (
+		latestErr error
+		result    []byte
+	)
 
-	src, err := ref.NewImageSource(ctx, systemContext)
-	if err != nil {
-		return errors.Wrapf(err, "error reading image %q", transports.ImageName(ref))
-	}
-	defer src.Close()
+	for _, ref := range refs {
+		logrus.Debugf("Testing reference %q for possible manifest", transports.ImageName(ref))
 
-	manifestBytes, manifestType, err := src.GetManifest(ctx, nil)
-	if err != nil {
-		return errors.Wrapf(err, "error loading manifest from image %q", transports.ImageName(ref))
+		src, err := ref.NewImageSource(ctx, systemContext)
+		if err != nil {
+			latestErr = errors.Wrapf(err, "error reading image %q", transports.ImageName(ref))
```

Comment:
Should we print out the latestErr for each failure?
```
if err != nil {
     if latestErr != nil {
             logurs.Errorf(latestErr)
    }
   latestErr = errors.Wrapf(err, "error reading image %q", transports.ImageName(ref))
}
```

Reply:
Yes makes sense, I prefer in having them as `Debug` messages since we only know if its an error after the loop has ended.

Reply:
That would be ok, but I would like to be able to retrieve the data rather then just drop it.

Reply:
Hm, we could concatenate / wrap the errors if that helps

Reply:
Wrapping it would be just fine.

Reply:
I added some logic to wrap the errors and return them in one bunch. I'm not sure if the resulting error message is helpful for users. PTAL :innocent: